packages/eslint-remote-tester/src/config/load.ts

```diff
@@ -1,44 +1,16 @@
-import { createRequire } from 'node:module';
-import type { Service, RegisterOptions } from 'ts-node';
-
-let registerer: Service | null = null;
-const require = createRequire(import.meta.url);
-
-/* istanbul ignore next */
-const interopRequireDefault = (obj: any): { default: any } =>
-    obj && obj.__esModule ? obj : { default: obj };
-
 /** @internal */
-export const loadTSConfig = (configPath: string) => {
-    try {
-        // eslint-disable-next-line @typescript-eslint/no-var-requires
-        registerer ||= require('ts-node').register({
-            moduleTypes: { '**/*.ts': 'cjs' },
-        } satisfies RegisterOptions) as Service;
-    } catch (e: any) {
-        if (e.code === 'MODULE_NOT_FOUND') {
-            throw new Error(
-                `'ts-node' is required for TypeScript configuration files. Make sure it is installed\nError: ${e.message}`
-            );
-        }
-
-        throw e;
-    }
-
-    registerer.enabled(true);
-
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    const configObject = interopRequireDefault(require(configPath)).default;
-
-    registerer.enabled(false);
+export const loadTSConfig = async (configPath: string) => {
+    const importx = await import('importx');
+    const config = await importx.import(configPath, import.meta.url);
```

Comment:
ESLint might start to use `jiti` in future: https://github.com/eslint/eslint/pull/18134

Switch to that one to be able to share the same peer dependency.

Reply:
There's also PR for `importx`: https://github.com/eslint/eslint/pull/18440

Maybe support both cases if ESLint doesn't merge those PRs yet.